core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt

```diff
@@ -304,49 +302,54 @@ fun DropdownTextField(
 /**
  * This function creates a phone number OutlinedTextField
  * @param modifier Set component modifier
- * @param value Current phone value
- * @param imeAction Signals the keyboard what type of action should be displayed
- * @param keyboardType Displays a phone keyboard
+ * @param value Current contact phone text value
  * @param isEnabled Establish the button is enabled and ready to use
- * @param showError This parameter determines whether the error is displayed or not
- * @param onPhoneChange Returns value typed
+ * @param isValid Indicates if the value introduced is valid
+ * @param supportTextError Indicates the error message when is not valid
+ * @param onPhoneChange Returns value typed, using high order functions
  *
  * @author Yaritza Moreno
+ * @author Jose Miguel Garcia Reyes (fixes)
  */
 @ExperimentalMaterial3Api
 @Composable
 fun LPPhoneNumberText(
     modifier: Modifier = Modifier,
     value: String,
-    imeAction: ImeAction = ImeAction.Next,
-    keyboardType: KeyboardType = KeyboardType.Phone,
     isEnabled: Boolean = true,
-    showError: Boolean = false,
+    isValid: Boolean,
+    supportTextError: String,
     onPhoneChange: (String) -> Unit
 ) {
-
-    //Text value for the TextField label
-    var text by remember { mutableStateOf("") }
-    val maxChar = 10
-    val focusManager = LocalFocusManager.current
-
+    var phoneNumberText by remember { mutableStateOf("") }
+    val phoneNumberTransformation = PhoneNumberTransformation()
+    val numbersOnlyExpression = Regex("^[0-9]*\$")
+    val maxCharactersAllowed = 10
     OutlinedTextField(
         modifier = modifier.fillMaxWidth(),
         label = { Text(text = stringResource(id = R.string.LPContactPhone)) },
+        placeholder = { Text(text = stringResource(id = R.string.LPContactPhone_mask)) },
         value = value,
-        onValueChange = {
-            text = it.take(maxChar)
-            if (it.length > maxChar) {
-                focusManager.moveFocus(FocusDirection.Down) // Or receive a lambda function
+        onValueChange = { it ->
+            phoneNumberText = if ( it.length > maxCharactersAllowed ) {
+                it.substring(0, it.length.coerceAtMost(10))
+            } else if ( !(it.matches(numbersOnlyExpression)) ) {
+                it.filter { it.isDigit() }
+            } else {
+                it
             }
             onPhoneChange(it)
         },
+        isError = isValid,
```

Comment:
If this `isValid` is false means it has a error?

Reply:
My apologies. It should be "isNotValid", I will re-name it to avoid confusion